File: services/backend/src/data/cliAnythingTemplates/python_skeleton/backend.py
import os
import shutil
import subprocess

def find_exe(name):
    """Find the executable for the target software."""
    path = shutil.which(name)
    if path:
        return path
    common_paths = [
        f'/usr/local/bin/{name}',
        f'/usr/bin/{name}',
        f'/opt/{name}/bin/{name}',
        os.path.expanduser(f'~/.local/bin/{name}'),
    ]
    for p in common_paths:
        if os.path.isfile(p) and os.access(p, os.X_OK):
            return p
    env_var = f'{name.upper()}_PATH'
    if os.environ.get(env_var):
        return os.environ[env_var]
    return None

def run_command(args, timeout=60, cwd=None):
    """Run a subprocess command and return structured result."""
    try:
        result = subprocess.run(
            args,
            capture_output=True,
            text=True,
            timeout=timeout,
            cwd=cwd,
        )
        return {
            'success': result.returncode == 0,
            'stdout': result.stdout,
            'stderr': result.stderr,
            'returncode': result.returncode,
        }
    except subprocess.TimeoutExpired:
        return {'success': False, 'error': f'Command timed out after {timeout}s'}
    except FileNotFoundError:
        return {'success': False, 'error': f'Executable not found: {args[0]}'}
    except Exception as e:
        return {'success': False, 'error': str(e)}

File: services/backend/src/data/cliAnythingTemplates/node_skeleton/backend.js
'use strict';

const { execFileSync } = require('child_process');
const path = require('path');
const fs = require('fs');

function findExe(name) {
  try {
    const cmd = process.platform === 'win32' ? 'where' : 'which';
    const result = require('child_process').execFileSync(cmd, [name], {
      encoding: 'utf-8', timeout: 5000, stdio: ['pipe', 'pipe', 'pipe'],
    }).trim().split(/\r?\n/)[0].trim();
    if (result) return result;
  } catch { /* not found */ }

  const commonPaths = [
    `/usr/local/bin/${name}`,
    `/usr/bin/${name}`,
    `/opt/${name}/bin/${name}`,
    path.join(require('os').homedir(), '.local', 'bin', name),
  ];
  for (const p of commonPaths) {
    try { if (fs.existsSync(p)) return p; } catch { /* skip */ }
  }

  const envVar = `${name.toUpperCase()}_PATH`;
  if (process.env[envVar]) return process.env[envVar];
  return null;
}

function runCommand(args, opts = {}) {
  const timeout = opts.timeout || 60000;
  const cwd = opts.cwd || undefined;
  try {
    const output = execFileSync(args[0], args.slice(1), {
      encoding: 'utf-8', timeout, cwd,
      stdio: ['pipe', 'pipe', 'pipe'],
      maxBuffer: 10 * 1024 * 1024,
    });
    return { success: true, stdout: output, stderr: '' };
  } catch (err) {
    return {
      success: false,
      error: err.message || 'command failed',
      stderr: err.stderr ? err.stderr.toString().slice(0, 2000) : '',
      exitCode: err.status || -1,
    };
  }
}

module.exports = { findExe, runCommand };
